app/controllers/register/trustees/individual/IndividualAddressListController.scala

```diff
@@ -46,29 +46,39 @@ class IndividualAddressListController @Inject()(override val appConfig: Frontend
 
   def onPageLoad(mode: Mode, index: Index, srn: Option[String]): Action[AnyContent] =
     (authenticate andThen getData(mode, srn) andThen allowAccess(srn) andThen requireData).async {
-    implicit request =>
-      viewmodel(mode, index, srn).right.map(get)
-  }
+      implicit request =>
+        viewmodel(mode, index, srn).fold {
+          Future.successful(Redirect(routes.IndividualPostCodeLookupController.onPageLoad(mode, index, srn)))
+        } {
+          get
+        }
+    }
 
   def onSubmit(mode: Mode, index: Index, srn: Option[String]): Action[AnyContent] = (authenticate andThen getData(mode, srn) andThen requireData).async {
     implicit request =>
-      viewmodel(mode, index, srn).right.map {
-        vm =>
-          post(vm, IndividualAddressListId(index), TrusteeAddressId(index), mode)
+      viewmodel(mode, index, srn).fold {
+        Future.successful(Redirect(routes.IndividualPostCodeLookupController.onPageLoad(mode, index, srn)))
+      } {
+        post(_, IndividualAddressListId(index), TrusteeAddressId(index), mode)
       }
   }
 
-  private def viewmodel(mode: Mode, index: Index, srn: Option[String])(implicit request: DataRequest[AnyContent]):
-  Either[Future[Result], AddressListViewModel] = {
-    (TrusteeDetailsId(index) and IndividualPostCodeLookupId(index)).retrieve.right.map {
-      case trusteeDetails ~ addresses => AddressListViewModel(
-        postCall = routes.IndividualAddressListController.onSubmit(mode, index, srn),
-        manualInputCall = routes.TrusteeAddressController.onPageLoad(mode, index, srn),
-        addresses = addresses,
-        subHeading = Some(Message(trusteeDetails.fullName)),
-        srn = srn
-      )
-    }.left.map(_ =>
-      Future.successful(Redirect(routes.IndividualPostCodeLookupController.onPageLoad(mode, index, srn))))
+  private def viewmodel(mode: Mode,
+                        index: Index,
+                        srn: Option[String]
+                       )(implicit request: DataRequest[AnyContent]): Option[AddressListViewModel] = {
+
+    (TrusteeDetailsId(index) and IndividualPostCodeLookupId(index)).retrieve.right.toOption.map {
```

Comment:
Please use a toggle here to use TrusteeNameId instead of TrusteeDetailsId when toggle is on

Reply:
Addressed as suggested